d2l-button-subtle.html

```diff
@@ -170,7 +170,8 @@
 			name$=[[name]]
 			type$=[[type]]>
 			<d2l-icon icon=[[icon]] class="d2l-button-subtle-icon"></d2l-icon>
-			<slot><span class="d2l-button-subtle-content">[[text]]</span></slot>
+			<span class="d2l-button-subtle-content">[[text]]</span>
```

Comment:
Before, you'd get the `span` if you didn't provide anything in a `slot`.  Now they'll always get it, this doesn't _seem_ like what we want.  What issue was this trying to fix again?  An actual issue with galen/shadycss itself?

@dbatiste is working on getting Perceptual Diff going, which I believe could replace the functionality galen is giving us.  If it is an issue with galen itself and that is actually coming, maybe not running that test in shadydom isn't a horrible solution?  Thoughts?

Reply:
Sounds good, I _think_ we'll only see an issue if someone sets the `text` attribute, but also puts something inside to go in the slot.  I don't know if this happens anywhere or if it makes sense, but the comment over the `text` property says "required"

Reply:
So it does say required on `text` just because we want to encourage the default usage to be using `text` rather than slot. Currently there should not be any cases that use both text and slot, since that would only use the slot, but it's possible. The slot is used in cases like MenuFlyout within the LMS, since it's styled like a subtle button but the handle can contain whatever (e.g., icons on both sides of the text) so I recommend testing this change there.

Reply:
I did some brief testing by matching the structure of the MenuFlyout dropdown - having the span inside the span outside the slot didn't have any noticeable difference.

---
As you pointed out @margaree, if both `slot` and `text` are used, the `text` will be ignored either way

From my understanding, I'm not sure which use cases would make sense with `slot/text` to both be set; If the user wants to put text and other content, they should only use the `slot`:

E.g., (icons on both sides)

<img width="794" alt="screen shot 2018-11-23 at 11 17 38 am" src="https://user-images.githubusercontent.com/6110668/48952777-6c953380-ef11-11e8-85b3-0d800bef5d03.png">

If the user were to try something like:

```html
<d2l-button-subtle text="All Threads" icon="d2l-tier1:chevron-down">
    <d2l-icon icon="d2l-tier1:chevron-gear"></d2l-icon>
</d2l-button-subtle>
```
only the gear would show.

In my opinion, I would expect that if I set `text`, it would show up regardless of if I use the `slot` or not
Any other thoughts? 
(Not sure if I'm moving away the discussion from the original problem or not 😛 )

Reply:
> In my opinion, I would expect that if I set text, it would show up regardless of if I use the slot or not

That's a good point, wanting everything set to do something to avoid the scenario we're in now (not knowing if this will break something).

Reply:
I think it makes the most sense to still recommend that it's an either or of slot vs text (maybe add a note to the readme?), with the strong recommendation being to use `text` and (if necessary) `icon` since that will get the design recommended subtle button style (http://design.d2l/components/buttons/) that the css exists for. The slot is really only there for existing weird LMS cases and I think, unless design patterns change, its use should be rare. We don't want to encourage it.

Thanks for testing - it seems that this shouldn't have impact on existing slot usages, so this change seems fine to me.

Reply:
That sounds good, I'll add a note to the README